src/main/kotlin/io/github/thewisenerd/linters/sidekt/rules/JsonIgnorePropertiesOnDataClass.kt

```diff
@@ -0,0 +1,88 @@
+package io.github.thewisenerd.linters.sidekt.rules
+
+import io.github.thewisenerd.linters.sidekt.helpers.Debugger
+import io.gitlab.arturbosch.detekt.api.*
+import org.jetbrains.kotlin.psi.KtClass
+
+/**
+ * This rule enforces the use of the `@JsonIgnoreProperties` annotation on all data classes within the codebase.
+ *
+ * Severity: Maintainability
+ * Debt: 5min
+ *
+ * Usage:
+ * JsonIgnorePropertiesOnDataClass:
+ *   active: true
+ *   excludes: "com.example.excluded, another.package"
+ */
+class JsonIgnorePropertiesOnDataClass(config: Config) : Rule(config) {
+
+    companion object {
+        private const val JSON_IGNORE_ANNOTATION = "JsonIgnoreProperties"
+        private const val IGNORE_UNKNOWN = "ignoreUnknown"
+    }
+
+    private val debugStream by lazy {
+        valueOrNull<String>("debug")?.let {
+            Debugger.getOutputStreamForDebugger(it)
+        }
+    }
+
+    private fun readConfig(key: String, initial: Set<String>? = null): Set<String> {
+        val result = initial?.toMutableSet() ?: mutableSetOf()
+        valueOrNull<ArrayList<String>>(key)?.let {
+            result.addAll(it)
+        }
+        return result
+    }
+
+    private val excludedPackages: Set<String> by lazy {
+        readConfig("excludes")
+    }
+
+    override val issue: Issue = Issue(
+        id = JsonIgnorePropertiesOnDataClass::class.java.simpleName,
+        severity = Severity.Maintainability,
+        description = "JsonIgnoreProperties(ignoreUnknown = true) is not annotated on the data class",
+        debt = Debt.FIVE_MINS
+    )
+
+    override fun visitClass(kclass: KtClass) {
+        super.visitClass(kclass)
+        val dbg = Debugger.make(JsonIgnorePropertiesOnDataClass::class.java.simpleName, debugStream)
+
+        val packageName = kclass.containingKtFile.packageFqName.asString()
+
+        // Check if the current class's package is in the excluded list
+        if (excludedPackages.any { packageName.startsWith(it) }) {
+            dbg.i("Package $packageName is excluded under JsonIgnoreProperties check")
+            return
+        }
+
+        // Check if the class is a data class
+        if (kclass.isData()) {
+            // Check if @JsonIgnoreProperties(ignoreUnknown = true) annotation exists
+            val hasJsonIgnoreProperties = kclass.annotationEntries.any { annotation ->
+                annotation.shortName?.asString() == JSON_IGNORE_ANNOTATION &&
```

Comment:
i'd rather you check the fully qualified name instead of the shortname

`<annotation> in setOf(<jackson.annotation>)`
i believe jackson2.x only has the _one_ import, so we can probably just do `<annotation> == <jackson.annotation>` until we have more to work with

we've had cases where we imported the annotation from the wrong package (tinkerpop because it shaded jackson) and that caused it to "not work"

cc/ @inolas93 if you remember any concrete examples

Reply:
ack, checking how to do this

Reply:
I am now checking the fully qualified name from the imports itself. Please check if this is correct.